*: add pingcap terms (#15870)
pingcap/docs-cn: resources/tidb-terms.md

---
title: TiDB 中英术语表
summary: 本文档介绍 TiDB 中英文术语对照表。
---

# TiDB 中英术语表

> | English | 中文 | 备注 | 参考 |

Reviewer: Suggest removing the fourth column for this whole table.

| Transparent Huge Pages (THP) | 透明大页 | A Linux memory management system |
| thread | 线程 |  |
| thread pool | 线程池 |  |
| throughput | 吞吐 |  |
| TiDB | TiDB |  |
| TiKV | TiKV |  |
| time out | 超时 |  |
| time series database | 时序数据库 |  |
| Time Series Database (TSDB) | 时序数据库 |  |
| time synchronization error | 时间同步误差 |  |
| Time To Live (TTL) | 存活时间 |  |
| timeliness | 时效性 |  |
| timestamp | 时间戳 |  |
| timestamp allocator | 时间戳分配器 |  |
| Timestamp Oracle (TSO) | Timestamp Oracle (TSO) |  |
| token bucket | 令牌桶 | An algorithm used in packet switched computer networks and telecommunications networks |
| tolerate | 容忍 |  |
| topology | 拓扑结构 |  |
| traffic flow control | 流控/流量控制 |  |
| transaction | 事务 |  |
| traverse | 遍历 |  |
| trie | 字典树 |  |
| trie | 单词查找树 |  |
| tutorial | 教程 |  |
| type | 类型 |  |
| type infer | 类型推导 |  |
| unbind | 解除（对服务、软件的）依赖，绑定，关联 |  |
| underlying table | 底层表 |  |
| unidirectional data flow | 单向数据流 |  |
| union | 联合体 |  |
| Universally Unique Identifier (UUID) | 通用唯一识别码 |  |
| unsigned integer | 无符号整数 |  |
| update | 修改（如：修改行） |  |
| user identification | 用户标识 |  |
| user privilege | 用户权限 |  |
| variable | 变量 |  |
| verification | 验证 |  |
| verify | 验证 |  |

> | Vertical Pod Autoscaling (VPA) | Pod 自动垂直扩容 | TiDB in Kubernetes |

Reviewer: ```suggestion
| Vertical Pod Autoscaling (VPA) | Pod 自动垂直扩容 | TiDB on Kubernetes |
```

---

> title: TiDB 术语表

Reviewer 2: ```suggestion
title: TiDB 中英术语表
```

> summary: 本文档介绍 TiDB 术语表

Reviewer 2: ```suggestion
summary: 本文档介绍 TiDB 中英文术语表。
```

---

> 

Reviewer 2: ```suggestion

本文档提供了 TiDB 中英文术语对照表。

```

| English | 中文 | 备注 | 参考 |

> |:---:|:---:|:---:|:---:|

Reviewer 2: ```suggestion
|:---|:---|:---|:---|
```

| in batches | 批量 | 动词短语 (verb phrase) |  |
| in full volume | 全量 |  |  |
| in-memory database | 内存数据库 |  |  |
| incremental backup | 增量备份 |  | <https://www.cockroachlabs.com/docs/stable/restore.html> |
| incremental import | 增量导入 |  |  |
| incremental synchronization | 增量同步 |  |  |
| incrementally | 增量地 |  |  |
| index engine | 索引引擎 |  |  |
| index join | 索引连接 |  |  |
| infer | 推导 |  |  |
| initialize | 初始化 |  |  |
| instance | 实例 |  |  |
| instantiate | 实例化 |  |  |
| int | 整数型 |  |  |
| interceptor | 拦截器 |  |  |
| interface | 接口 |  |  |
| intermediate state | 中间状态 |  |  |
| internal mechanism | 内部机制 |  |  |
| Internationalization (I18N) | 国际化 |  |  |
| Internet Data Center (IDC) | 互联网数据中心 |  |  |
| intrusive | 有侵入性的 |  |  |
| IOPS (Input/Output Operations Per Second) | IOPS | 每秒输入输出操作，一般写英文形式 |  |
| Isolated storage | 独立存储 |  |  |
| isolation | 隔离 |  |  |
| isolation level | 隔离级别 |  |  |
| ISP (Internet Service Provider) | 互联网服务提供商 |  | <https://en.wikipedia.org/wiki/Internet_service_provider> |
| JDBC (Java DataBase Connectivity) | Java 数据库连接 |  |  |
| kernel version | 内核版本 |  |  |
| key-value pair(s) / KV pair(s) | 键值对 |  |  |
| keyword | 关键字 |  |  |

> | Kubernetes (K8s) | <https://kubernetes.io/docs/concepts/overview/what-is-kubernetes/#what-does-kubernetes-mean-k8s> |

Reviewer 2: ```suggestion
| Kubernetes (K8s) | Kubernetes (K8s) |  | <https://kubernetes.io/docs/concepts/overview/what-is-kubernetes/#what-does-kubernetes-mean-k8s> |
```

| privilege | 权限 |  |  |
| production environment | 生产环境 |  |  |
| public cloud | 公有云 |  |  |
| public key | 公钥 |  |  |
| pure virtual function | 纯虚函数 |  |  |
| push | 推送 |  |  |
| push down | 下推 | 动词短语 (verb phrase) |  |
| pushdown | 下推的 | 形容词 (adj.) |  |
| query | 查询 |  |  |
| query by returning to the table (not a fixed term) | 回表查询 |  |  https://odetocode.com/articles/237.aspx |   |  |
| query tree | 查询树 |  |  |
| Raft | Raft |  |  |
| Raft consensus algorithm | Raft 一致性算法 |  |  |
| RAID (Redundant Array of Independent Disks) | 独立冗余磁盘阵列 |  |  |
| random delay | 随机延迟 |  |  |
| range query | 范围查询 |  |  |
| RBO (rule based optimization) | 基于规则的优化 |  |  |
| RDBMS: Relational Database Management System | 关系型数据库管理系统 |  |  |
| read | 读 |  |  |
| reads and writes | 读写 |  |  |
| real-time replication delay | 实时同步延迟 |  |  |
| recursion | 递归 |  |  |
| reduction operation | 归约操作 |  |  |
| refactor | 重构 |  |  |
| Region | Region |  |  |
| regular expression | 正则表达式 |  |  |
| remote disaster recovery | 异地灾备 |  |  |
| renderer | 渲染器 |  |  |
| repair data | 修复数据 |  |  |
| replica | 副本 |  |  |
| replicator | 复制因子 |  |  |
| repository | 仓库 |  |  |
| request | 请求 |  |  |
| restart | 重启 |  |  |

> | restore | 恢复 | 名词 (noun)、动词 (verb) | https://docs.microsoft.com/en-us/sql/relational-databases/backup-restore/back-up-and-restore-of-sql-server-databases?view=sql-server-ver15 |

Author: ```suggestion
| restore | 恢复 | 可以作为名词 (noun) 或动词 (verb) 使用 | https://docs.microsoft.com/en-us/sql/relational-databases/backup-restore/back-up-and-restore-of-sql-server-databases?view=sql-server-ver15 |
```

---

> title: PingCAP 术语表
> summary: 本文档介绍 PingCAP 术语表

Author: ```suggestion
title: TiDB 术语表
summary: 本文档介绍 TiDB 术语表
```

---

> | English | 中文 | Comment | Reference |

Reviewer 2: ```suggestion
| English | 中文 | 备注 | 参考 |
```

|:---:|:---:|:---:|:---:|

> | 2-phase commit (2PC) | 2阶段提交 |  |  |

Author: ```suggestion
| 2-phase commit (2PC) | 2 阶段提交 |  |  |
```

| abort | 退出 |  |  |
| abstract | 抽象 |  |  |
| abstract syntax tree (AST) | 抽象语法树 |  |  |
| abstraction | 抽象语法树 |  |  |
| access | 访问 |  |  |
| account | 账户 |  |  |
| accuracy | 准确性 |  |  |

> | ACID properties of   transactions | 事务的 ACID 特性 |  | https://www.ibm.com/support/knowledgecenter/en/SSGMCP_5.3.0/com.ibm.cics.ts.productoverview.doc/concepts/acid.html |

Reviewer 2: ```suggestion
| ACID properties of transactions | 事务的 ACID 特性 |  | https://www.ibm.com/support/knowledgecenter/en/SSGMCP_5.3.0/com.ibm.cics.ts.productoverview.doc/concepts/acid.html |
```

| aggregate expression | 聚合表达式 |  | https://www.ibm.com/support/knowledgecenter/en/SSGU8G_12.1.0/com.ibm.sqls.doc/ids_sqs_0189.htm |
| aggregate function | 聚合函数 |  |  |
| alert | 报警 |  |  |
| algorithm | 算法 |  |  |
| anonymous | 匿名 |  |  |
| appendix | 附录 |  |  |
| architecture | 架构 |  |  |
| arithmetic operation | 数值运算/算术运算 |  |  |
| ascending order | 正序 |  |  |
| assignment expression | 赋值表达式 |  |  |
| assignment operation | 赋值运算 |  |  |
| authorize | 授予权限 |  |  |
| auto increment | 自增 |  |  |
| auto-concurrent/auto-concurrency | 自动线程数并行 |  |  |
| auto-failover | 故障自恢复 |  |  |
| auto-increment primary key | 自增主键 |  |  |
| availability | 可用性 |  |  |
| availability zone | 可用区 |  | https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using-regions-availability-zones.html#concepts-regions-availability-zones |
| backup | 备份 |  |  |
| bastion | 堡垒机 |  |  |
| binary literal | 二进制字面值 |  |  |

> | bit | 位 |  | in cryptography |

Reviewer 2: ```suggestion
| bit | 位 |  |  |
```

| Bloom filter | 布隆过滤器 |  | https://en.wikipedia.org/wiki/Bloom_filter |
| boolean operation | 布尔运算 |  |  |
| bootstrap | 启动\引导程序 |  |  |
| breakpoint | 断点 |  |  |
| breakpoint resume | 断点续传 |  |  |
| business continuity | 业务可持续性 |  |  |
| cache | 缓存 |  |  |
| cache miss | 缓存未命中 |  |  |
| canary test | 灰度测试 |  |  |
| Cartesian product | 笛卡尓积 |  |  |
| case-sensitive | 大小写敏感 |  |  |
| CBO (cost based optimization) | 基于代价的优化 |  |  |

> | certificate signing request   (CSR) | 证书签名请求 |  |  |
> | Custom Resource Definition   (CRD) | 自定义资源类型 |  |  |

Reviewer 2: ```suggestion
| certificate signing request (CSR) | 证书签名请求 |  |  |
| Custom Resource Definition (CRD) | 自定义资源类型 |  |  |
```

| checksum | 校验和 |  |  |
| circular dependency | 循环依赖 |  |  |
| class | 类 |  |  |
| client | 客户端 |  |  |

> | client library | 客户端库 | added by Xiaozhen Liu |  |

Author: ```suggestion
| client library | 客户端库 |   |  |
```

| cluster | 集群 |  |  |
| cluster functionality | 集群功能 |  |  |
| clustered index | 聚簇索引 |  |  |
| coarse-grained | 粗粒度的 |  |  |
| code comment | 代码注释 |  |  |
| code instrumentation | 代码插桩 |  | https://www.itprotoday.com/development-techniques-and-management/code-instrumentation-techniques-depth-coverage-tracing |
| collation | 排序/排序规则 |  | https://github.com/pingcap/docs-cn/blob/master/op-guide/tune-tikv.md |
| column family (CF) | 列族 |  | https://github.com/pingcap/docs-cn/blob/master/op-guide/tune-tikv.md |
| column pruning | 列裁剪 |  |  |
| column store | 列存 |  |  |
| column-oriented DBMS | 列式存储数据库 |  |  |
| columnar storage engine | 列式存储引擎 |  |  |
| combinatorial test | 组合测试 |  |  |

> | combined table | 总表 | “分表”和“总表”的英文翻译由李琳咨询王相后提供 |  |

Author: ```suggestion
| combined table | 总表 |   |  |
```

| configuration item | 配置项 |  |  |
| conflict | 冲突 |  |  |
| consistency | 一致性 |  |  |
| constant | 常数，常量 |  |  |
| constant folding | 常量折叠 |  |  |
| constraint | 约束 |  |  |
| container technology | 容器技术 |  |  |
| Control Machine | 中控机 |  | https://docs.ansible.com/ansible/latest/installation_guide/intro_installation.html#control-machine-requirements |
| converge | 收敛 |  |  |
| Coprocessor | Coprocessor | 无需翻译；是否大写首字母看上下文 |  |
| correctness | 正确性 |  |  |
| correlated subquery | 关联子查询 |  | https://en.wikipedia.org/wiki/Correlated_subquery |
| corresponding operation | 对应操作 |  |  |
| CPU-bound | CPU 密集型 |  |  |
| cross-datacenter | 跨数据中心 |  |  |
| cross-row transaction | 跨行事务 |  |  |
| custom schedulers | 自定义调度器 |  |  |
| daily full synchronization | 每日全量同步 |  |  |
| data abstraction | 数据抽象 |  |  |
| data backfilling | 数据回填 |  |  |
| data block | 数据块 |  |  |
| data directory | 数据路径 |  |  |
| data dissemination | 数据分发 |  |  |
| data engine | 数据引擎 |  |  |
| data hub | 数据中台 |  |  |
| data race | 数据竞争 |  |  |
| database administrator (DBA) | 数据库管理员 |  |  |
| database kernel | 数据库内核 | 内核是 kernel，不是 core |  |
| datacenter | 数据中心 |  |  |
| dataflow | 数据流 |  |  |

> | decompress | 解压缩 |  | The difference between "decompress" and   "uncompress": https://english.stackexchange.com/questions/56480/difference-between-uncompress-and-decompress |

Reviewer 2: ```suggestion
| decompress | 解压缩 |  | The difference between "decompress" and "uncompress": https://english.stackexchange.com/questions/56480/difference-between-uncompress-and-decompress |
```

> | decompress | 解压缩 |  | The difference between "decompress" and   "uncompress": https://english.stackexchange.com/questions/56480/difference-between-uncompress-and-decompress |

Author: ```suggestion
| decompress | 解压缩 |  | The difference between "decompress" and "uncompress": https://english.stackexchange.com/questions/56480/difference-between-uncompress-and-decompress |
```

> | decompress | 解压缩 |  | The difference between "decompress" and   "uncompress":   https://english.stackexchange.com/questions/56480/difference-between-uncompress-and-decompress |

Author: ```suggestion
| decompress | 解压缩 |  | The difference between "decompress" and   "uncompress": https://english.stackexchange.com/questions/56480/difference-between-uncompress-and-decompress |
```

> | database kernel | 数据库内核 | 内核是kernel，不是core |  |  |  |

Reviewer 2: ```suggestion
| database kernel | 数据库内核 | 内核是 kernel，不是 core |  |  |  |
```

| delta import | 增量导入 |  |  |
| denpendency/dependencies | 依赖包 |  |  |
| dependency | 依赖关系 |  |  |
| deploy | 部署 |  |  |
| deployment | 部署 |  |  |
| destroy the cluster | 销毁集群 |  |  |
| directory | 目录 |  |  |
| disaster recovery | 容灾 |  | https://en.wikipedia.org/wiki/Disaster_recovery |
| disparate data source | 异构数据源 |  |  |
| distributed database | 分布式数据库 |  |  |

> | DML (Data Manipulation   Language) | 数据操纵语言 |  |  |

Author: ```suggestion
| DML (Data Manipulation Language) | 数据操纵语言 |  |  |
```

| driving table | 驱动表 |  |  |
| dummy data | 假数据 |  |  |
| dynamic | 动态 |  |  |
| dynamic programming algorithm | 动态规划算法 |  | https://zh.wikipedia.org/wiki/%E5%8A%A8%E6%80%81%E8%A7%84%E5%88%92 |
| effective range | 生效范围 |  |  |
| earlier version | 以下版本 |  |  |
| edit | 修改（如：ini 文件） |  |  |
| effective range | 生效范围 |  |  |
| elect | 选举 |  |  |
| engine file | 引擎文件 |  |  |
| environment variable | 环境变量 |  |  |
| equivalent/equal condition | 等值条件 |  |  |
| escape character | 转义字符 |  | http://www.webopedia.com/TERM/E/escape_character.html |
| exact match | 精确匹配 |  |  |
| exception | 异常 |  |  |
| execute | 执行 |  |  |
| experimental feature | 实验特性 |  |  |
| expire | 过期 |  |  |
| explicit transaction | 显式事务 |  |  |
| expression | 表达式 |  |  |
| extendible hashing | 可扩充散列 |  |  |
| external interface | 对外接口 |  |  |
| eventual consistency | 最终一致性 |  | https://en.wikipedia.org/wiki/Eventual_consistency |
| fan trap | 扇形陷阱 |  |  |
| fanout | 扇出 |  |  |
| fault-tolerant design | 容错误设计 |  |  |
| fetch data | 捞数据 |  |  |
| field | 字段 |  |  |
| file descriptor | 文件描述符 |  |  |
| file handle | 文件句柄 |  |  |
| fill-factor | 填充因子 |  |  |
| fine-grained | 细粒度 |  |  |

> | flag | (命令行)标记/参数 |  |  |

Author: ```suggestion
| flag | （命令行）标记/参数 |  |  |
```

> | float |  |  |  |

Reviewer 2: ```suggestion
| float | float | |  |  |
```

| Follower | Follower | A role in the Raft consensus algorithm |  |
| force upgrade | 强制升级 |  |  |
| foundation class | 基础类库 |  |  |
| framework | 框架 |  |  |
| full backup | 全量备份 |  | https://www.cockroachlabs.com/docs/stable/restore.html |
| full compact / compaction | 全量压缩 |  |  |
| full import | 全量导入 |  |  |
| full match | 全匹配 |  |  |
| full table scan | 全表扫描 |  |  |
| function | 函数 |  |  |
| function signature | 函数签名 |  |  |
| functional abstraction | 功能抽象 |  | http://www.wendangku.net/doc/9bb5ba5db90d6c85ed3ac614-4.html |
| fuzzy matching | 模糊匹配 |  |  |
| Geo-distributed active-active | 异地双活 |  | https://redislabs.com/redis-enterprise-documentation/concepts-architecture/intercluster-replication/ |
| Geo-distributed multi-active | 异地多活 |  | https://www.ibm.com/support/knowledgecenter/en/SSEPGG_10.1.0/com.ibm.db2.luw.admin.sd.doc/doc/c0058742.html |
| global | 全局 |  |  |
| global privileges | 全局权限 |  |  |

> | goroutine | 协程 | It also can be "coroutine". But it is   suggested using "goroutine" by the TiDB Tools team. |  |

Author: ```suggestion
| goroutine | 协程 | It also can be "coroutine". But it is suggested using "goroutine" by the TiDB Tools team. |  |
```

| graceful | 优雅的 |  |  |
| granularity | 粒度 |  |  |
| grid computing | 网格计算 |  |  |

> | GTID (Global Transaction   Identifier) | 全局事务标识符 |  |  |

Author: ```suggestion
| GTID (Global Transaction Identifier) | 全局事务标识符 |  |  |
```

| HA policy | 高可用策略 |  |  |
| handle | 句柄 |  |  |
| hang | 假死 |  |  |
| hard limit | 硬限制 |  |  |
| hash function | 散列函数 |  |  |
| hash join | 哈希连接 |  |  |
| hash partitioned table | hash 分区表 |  | https://docs.oracle.com/database/121/VLDBG/GUID-04E6C6EF-CE27-4DE9-B5B1-D23010E4A356.htm |
| high availability | 高可用 |  |  |

> | high concurrency | 高并发 | noun phrase |  |
> | high concurrent | 高并发 | adj phrase |  |

Author: ```suggestion
| high concurrency | 高并发 | 名词短语 (noun phrase) |  |
| high concurrent | 高并发 | 形容词短语 (adj phrase) |  |
```

> | high concurrency | 高并发 | noun phrase |  |
> | high concurrent | 高并发 | adj phrase |  |

Reviewer 2: ```suggestion
| high concurrency | 高并发 | noun phrase |  |
| high concurrent | 高并发 | adj phrase |  |
```
```suggestion
| high concurrency | 高并发 | 名词 |  |
| high concurrent | 高并发 | 形容词 |  |
```

| horizontal scalability | 水平扩展 |  |  |
| host | 主机 |  |  |
| hot data | 热点数据 |  |  |
| hotfix | 热修复 |  |  |

> | HPA (Horizontal Pod   Autoscaling) | 自动水平扩容 | TiDB in Kubernetes | https://cloud.google.com/blog/products/gcp/beyond-cpu-horizontal-pod-autoscaling-comes-to-google-kubernetes-engine |

Author: ```suggestion
| HPA (Horizontal Pod Autoscaling) | 自动水平扩容 | TiDB in Kubernetes | https://cloud.google.com/blog/products/gcp/beyond-cpu-horizontal-pod-autoscaling-comes-to-google-kubernetes-engine |
```

> | HTAP (Hybrid Transactional and   Analytical Processing) | HTAP (Hybrid Transactional and Analytical Processing) | 一般写英文形式 |  |

Author: ```suggestion
| HTAP (Hybrid Transactional and Analytical Processing) | HTAP (Hybrid Transactional and Analytical Processing) | 一般写英文形式 |  |
```

| hybrid deployment | 混合部署 |  |  |
| idempotent (idempotence) | 幂等 |  |  |
| image | 镜像 |  | https://docs.docker.com/engine/reference/commandline/image/ |

> | image tag error | 镜像tag错误 |  |  |

Author: ```suggestion
| image tag error | 镜像 tag 错误 |  |  |
```

| implement | 实现 |  |  |
| implementation | 实现 |  |  |
| implicit transaction | 隐式事务 |  |  |

> | in batches | 批量 | verb phrase |  |

Author: ```suggestion
| in batches | 批量 | 动词短语 (verb phrase) |  |
```

| latch contention | 锁竞争 |  |  |
| latency | 延时 |  |  |
| later version | 以上版本 |  | https://github.com/pingcap/docs/blob/master/sql/encrypted-connections.md |
| Leader | Leader | A role in the Raft consensus algorithm |  |
| leak | 泄漏 |  |  |
| least recently used (LRU) | 近期最少使用法 |  |  |
| left associative | 左结合 |  |  |
| left outer join | 左外连接 |  |  |
| level | 级别 |  |  |
| lifecycle | 生命周期 |  |  |
| limitation | 限制 |  |  |
| linear hashing | 线性散列 |  |  |
| listening | （服务）监听 |  |  |
| listening address | 监听地址 |  |  |
| load | 负载 |  |  |
| load | 加载 |  |  |
| Local Persist Volume(s)/LPV(s) | 本地持久化卷 |  |  |
| locality | 局部性 |  |  |
| Localization (L10N) | 本地化 |  |  |
| lock | 锁 |  |  |
| lock contention | 锁争用 |  |  |
| log file | 日志文件 |  |  |
| log level | 日志等级 |  |  |
| logic | 逻辑 |  |  |
| logic isolation | 逻辑隔离 |  |  |
| logical operation | 逻辑运算 |  |  |
| logical plan | 逻辑计划 |  |  |
| long transaction | 长事务 |  | https://www.ibm.com/developerworks/cn/data/library/techarticles/dm-1001haodh/index.html |
| machine code | 机器码 |  |  |
| massive data | 海量数据 | 不是 big data（大数据） |  |

> | Massively Parallel Processing   (MPP) | 大规模并行处理 |  |  |

Author: ```suggestion
| Massively Parallel Processing (MPP) | 大规模并行处理 |  |  |
```

| master | 主库 |  |  |
| materialized view | 物化视图 |  |  |
| memory | 内存 |  |  |
| merge join | 合并连接 |  |  |
| metadata | 元信息 |  |  |
| method | 方法 |  |  |
| method invocation | 方法调用 |  |  |
| metrics | 统计/指标（信息） |  |  |
| microservice | 微服务 |  |  |
| middleware | 中间件 |  |  |
| migration | 迁移 |  |  |
| module call | 模块调用 |  |  |
| monotone increasing | 单调递增 |  |  |
| mount option(s) | 挂载参数 |  |  |
| mount/mounting | 挂载 |  |  |
| multi nodes | 多节点 |  |  |
| multi-tenancy | 多租户技术 |  |  |
| mutation | 变异 |  | http://www.orafaq.com/wiki/Mutating_table       https://blog.csdn.net/zhpsam109/article/details/97335 |
| mutex | 互斥锁 | Mutual exclusion | https://zh.wikipedia.org/wiki/%E4%BA%92%E6%96%A5%E9%94%81 |

> | MVCC: multi-version   concurrency control | 多版本并发控制 |  |  |

Author: ```suggestion
| MVCC: multi-version concurrency control | 多版本并发控制 |  |  |
```

| MySQL syntax | MySQL 语法 |  |  |
| namespace | 命名空间 |  |  |
| native code | 原生码 |  |  |
| nest loop | 嵌套循环 |  |  |
| node | 节点 |  |  |
| non-unique index | 非唯一索引 |  |  |
| null | 空值 |  |  |
| object abstraction | 对象抽象 |  |  |
| offset | 偏移 |  |  |

> | OLAP: Online Analytical   Processing | 在线分析处理 |  |  |
> | OLTP: Online Transactional   Processing | 在线事务处理 |  |  |

Author: ```suggestion
| OLAP: Online Analytical Processing | 在线分析处理 |  |  |
| OLTP: Online Transactional Processing | 在线事务处理 |  |  |
```

| one-to-many relationship | 一对多关系 |  |  |
| open addressing hashing | 开地址法 |  |  |

> | open source | 开源 | noun |  |

Author: ```suggestion
| open source | 开源 | 名词 (noun) |  |
```

> | open-source | 开源 | adj, verb |  |

Author: ```suggestion
| open-source | 开源 | 形容词 (adj)、动词 (verb) |  |
```

> | open-source | 开源 | adj, verb |  |

Reviewer 2: ```suggestion
| open source | 开源 | 名词 |  |
| open-source | 开源 | 形容词 |  |
```

> | massive data | 海量数据 | 不是big data（大数据） |  |  |  |

Reviewer 2: ```suggestion
| massive data | 海量数据 | 不是 big data（大数据） |  |  |  |
```

| operating environment | 运行环境 |  |  |
| operation and administer | 运维 |  | https://en.wikipedia.org/wiki/Operations,_administration_and_management |
| operational data store (ODS) | 操作数据存储 |  |  |
| operational status | 运行状态 |  |  |
| operator | 运算符/操作符/算子 |  |  |
| operator precedence | 运算符优先级 |  |  |
| optimistic locking | 乐观锁 |  |  |
| optimistic transaction model | 乐观事务模型 |  |  |
| optimizer | 优化器 |  |  |

> | ORM (Object Relational   Mapping) | 对象关系映射 |  |  |

Author: ```suggestion
| ORM (Object Relational Mapping) | 对象关系映射 |  |  |
```

| out of memory (OOM) | 内存溢出 |  | https://en.wikipedia.org/wiki/Out_of_memory |
| outer join | 外连接 |  |  |
| outer join simplification | 外连化简 |  |  |
| overhead | 开销 |  |  |
| page fault | 页失效 |  |  |
| paging | 分页 |  |  |
| parallel replication | 并行复制 |  |  |
| parameter | 参数 |  |  |
| parse | 解析 |  |  |

> | partitioned table/partition   table | 分区表 |  | https://docs.oracle.com/database/121/VLDBG/GUID-6D369646-16AF-487B-BF32-5F6569D27C8A.htm#VLDBG1046       https://en.wikipedia.org/wiki/Partition_table |

Author: ```suggestion
| partitioned table/partition table | 分区表 |  | https://docs.oracle.com/database/121/VLDBG/GUID-6D369646-16AF-487B-BF32-5F6569D27C8A.htm#VLDBG1046       https://en.wikipedia.org/wiki/Partition_table |
```

| performance | 性能 |  |  |
| performance jitter | 性能抖动 |  |  |
| performance overhead | 性能开销 |  |  |
| performance tuning | 性能调优 |  |  |

> | persist (data) | 持久化存储（数据） | persist 与一般的   save（保存）不一样，持久化存储是将数据保存到硬盘中，使得在应用程序或机器重启后可以继续访问之前保存的数据。 | https://www.jianshu.com/p/6ce0864d2845 |

Author: ```suggestion
| persist (data) | 持久化存储（数据） | persist 与一般的 save（保存）不一样，持久化存储是将数据保存到硬盘中，使得在应用程序或机器重启后可以继续访问之前保存的数据。 | https://www.jianshu.com/p/6ce0864d2845 |
```

| persistent data | 持久化数据 |  |  |
| pessimistic locking | 悲观锁 |  |  |
| physical plan | 物理计划 |  |  |
| PoC (proof of concept) | 概念验证 |  |  |
| point query | 点查询 |  |  |
| populate | 填充 |  |  |
| port | 端口 |  |  |

> | precheck | 前置检查 | TiDB 文档里统一译成这个，不译成预检查 |  |

Author: ```suggestion
| precheck | 前置检查 | TiDB 文档里统一译成“前置检查”，不译成”预检查” |  |
```

| prefix match | 前缀匹配 |  |  |

> | primitive | 原语 | Kubernetes 中的概念 | InfoQ：Kubernetes 的核心概念讲解 |

Author: ```suggestion
| primitive | 原语 | Kubernetes 中的概念 | [InfoQ：Kubernetes 的核心概念讲解](https://www.infoq.cn/article/69dzo9zwe5y3ca*5quqk) |
```

| private cloud | 私有云 |  |  |
| private key | 私钥 |  |  |
| privilege | 权限 |  |  |
| production environment | 生产环境 |  |  |
| public cloud | 公有云 |  |  |
| public key | 公钥 |  |  |
| pure virtual function | 纯虚函数 |  |  |
| push | 推送 |  |  |

> | push down | 下推 | verb phrase |  |
> | pushdown | 下推的 | adj. |  |

Author: ```suggestion
| push down | 下推 | 动词短语 (verb phrase) |  |
| pushdown | 下推的 | 形容词 (adj.) |  |
```

> | pushdown | 下推的 | adj. |  |

Reviewer 2: ```suggestion
| push down | 下推 | 动词 |  |
| pushdown | 下推的 | 形容词 |  |
```

| query | 查询 |  |  |

> | query by returning to the   table (not a fixed term) | 回表查询 |  |  https://odetocode.com/articles/237.aspx |   |  |

Author: ```suggestion
| query by returning to the table (not a fixed term) | 回表查询 |  |  https://odetocode.com/articles/237.aspx |   |  |
```

| query tree | 查询树 |  |  |
| Raft | Raft |  |  |
| Raft consensus algorithm | Raft 一致性算法 |  |  |

> | RAID (Redundant Array of   Independent Disks) | 独立冗余磁盘阵列 |  |  |

Author: ```suggestion
| RAID (Redundant Array of Independent Disks) | 独立冗余磁盘阵列 |  |  |
```

| random delay | 随机延迟 |  |  |
| range query | 范围查询 |  |  |
| RBO (rule based optimization) | 基于规则的优化 |  |  |

> | RDBMS: Relational Database   Management System | 关系型数据库管理系统 |  |  |

Author: ```suggestion
| RDBMS: Relational Database Management System | 关系型数据库管理系统 |  |  |
```

| read | 读 |  |  |
| reads and writes | 读写 |  |  |
| real-time replication delay | 实时同步延迟 |  |  |
| recursion | 递归 |  |  |
| reduction operation | 归约操作 |  |  |
| refactor | 重构 |  |  |
| Region | Region |  |  |
| regular expression | 正则表达式 |  |  |
| remote disaster recovery | 异地灾备 |  |  |
| renderer | 渲染器 |  |  |
| repair data | 修复数据 |  |  |
| replica | 副本 |  |  |
| replicator | 复制因子 |  |  |
| repository | 仓库 |  |  |
| request | 请求 |  |  |
| restart | 重启 |  |  |

> | restore | 恢复 | noun, verb | https://docs.microsoft.com/en-us/sql/relational-databases/backup-restore/back-up-and-restore-of-sql-server-databases?view=sql-server-ver15 |

Author: ```suggestion
| restore | 恢复 | 名词 (noun)、动词 (verb) | https://docs.microsoft.com/en-us/sql/relational-databases/backup-restore/back-up-and-restore-of-sql-server-databases?view=sql-server-ver15 |
```

> | restore | 恢复 | noun, verb | https://docs.microsoft.com/en-us/sql/relational-databases/backup-restore/back-up-and-restore-of-sql-server-databases?view=sql-server-ver15 |

Reviewer 2: ```suggestion
| restore | 恢复 | 可以作为名词或动词使用 | https://docs.microsoft.com/en-us/sql/relational-databases/backup-restore/back-up-and-restore-of-sql-server-databases?view=sql-server-ver15 |
```

> | query by returning to the   table (not a fixed term) | 回表查询 | ref: https://odetocode.com/articles/237.aspx |  |  |  |

Reviewer 2: ```suggestion
| query by returning to the   table (not a fixed term) | 回表查询 |  |  https://odetocode.com/articles/237.aspx |   |  |
```

| scalibility | 扩展性 |  |  |
| scenario | 场景 |  |  |
| schedule | 调度 |  |  |
| scope | 作用域 |  |  |
| secondary indices | 次级索引 |  |  |
| seed | 数据填充 |  |  |
| sequential flooding | 顺序泛洪 |  |  |
| service | 服务 |  |  |
| session | 会话 |  | https://blog.csdn.net/jimsonhappy/article/details/54707694 |
| shard key | 分片键 |  |  |
| shard(s) | 分片（名词） | 比如“删除一个分片”，“SHARD_ROW_ID_BITS = 4 代表 16 个分片”等，此时用 shard(s)。 | http://www.mongoing.com/docs/tutorial/remove-shards-from-cluster.html |
| sharded table | 分表 |  | https://docs.oracle.com/en/database/oracle/oracle-database/12.2/admin/sharding-schema-design.html#GUID-1B8DAEE1-A662-4516-9EA9-04FF711211C3 |
| sharding | 分片（动词或动名词） |  |  |
| sign bit | 符号位 |  | https://en.wikipedia.org/wiki/Sign_bit |
| simulate | 模拟 |  |  |
| single-instance | 单机 |  |  |
| skip list | 跳跃表 |  |  |
| slave | 从库 |  |  |
| slow requery | 慢查询 |  |  |
| snapshot | 快照 |  |  |
| snapshot isolation (SI) | 快照隔离 |  | https://docs.microsoft.com/en-us/dotnet/framework/data/adonet/sql/snapshot-isolation-in-sql-server |
| soft limit | 软限制 |  |  |
| sort | 排序 |  |  |
| sorted | 有序的 |  |  |
| source end | 源端 |  |  |
| split | 切分 |  |  |
| SPOF: single point of failure | 单点故障 |  |  |

> | SST (Static Sorted Table) | https://github.com/facebook/rocksdb/wiki/A-Tutorial-of-RocksDB-SST-formats |

Reviewer 2: ```suggestion
| SST (Static Sorted Table) | 静态排序表文件 (SST) |  | https://github.com/facebook/rocksdb/wiki/A-Tutorial-of-RocksDB-SST-formats  |
```

| stack | 堆栈 |  |  |
| startup parameter | 启动参数 |  |  |
| stateful | 有状态的 |  |  |
| stateless | 无状态的 |  |  |
| statement | 语句 |  |  |

> | stop | 下线 |  |  |

Reviewer 2: ```suggestion
```

| storage capacity | 存储能力 |  |  |
| storage engine | 存储引擎 |  |  |
| stored procedure | 存储过程 |  | https://en.wikipedia.org/wiki/Stored_procedure |
| stream computing | 流计算 |  |  |
| stress test | 压力测试 |  |  |
| string | 字符串 |  |  |
| strong consistency | 强一致性 |  |  |
| struct | 结构体 |  |  |
| subquery | 子查询 |  |  |
| super class | 超类 |  |  |
| synchronizing/syncing | 同步 |  |  |

> | Syntactic sugar | 语法糖 | In computer science, syntactic sugar is syntax   within a programming language that is designed to make things easier to read   or to express. It makes the language "sweeter" for human use:   things can be expressed more clearly, more concisely, or in an alternative   style that some may prefer. |  |

Author: ```suggestion
| Syntactic sugar | 语法糖 | In computer science, syntactic sugar is syntax within a programming language that is designed to make things easier to read or to express. It makes the language "sweeter" for human use: things can be expressed more clearly, more concisely, or in an alternative style that some may prefer. |  |
```

| syntax | 语法 |  |  |
| system variable | 系统变量 |  |  |

> | THP (Transparent Huge Pages) | 透明大页 | a Linux memory management system | https://docs.pingcap.com/zh/tidb/dev/tune-operating-system#%E5%86%85%E5%AD%98%E9%80%8F%E6%98%8E%E5%A4%A7%E9%A1%B5 |

Author: ```suggestion
| THP (Transparent Huge Pages) | 透明大页 | A Linux memory management system | https://docs.pingcap.com/zh/tidb/dev/tune-operating-system#%E5%86%85%E5%AD%98%E9%80%8F%E6%98%8E%E5%A4%A7%E9%A1%B5 |
```

| table schema | 表结构 |  |  |
| target machine | （部署）目标机器 |  | http://docs.ansible.com/ansible/latest/intro_inventory.html |
| technology stack | 技术栈 |  |  |
| test | 测试 |  |  |
| the configuration file | 配置文件 |  |  |
| thread | 线程 |  |  |
| thread pool | 线程池 |  |  |
| throughput | 吞吐 |  |  |
| TiDB | TiDB |  |  |
| TiKV | TiKV |  |  |
| time out | 超时 |  |  |
| time series database | 时序数据库 |  |  |
| Time Series Database (TSDB) | 时序数据库 |  |  |
| time synchronization error | 时间同步误差 |  |  |
| timeliness | 时效性 |  |  |
| timestamp | 时间戳 |  |  |
| timestamp allocator | 时间戳分配器 |  |  |
| Timestamp Oracle | TSO |  |  |

> | token bucket | 令牌桶 | an algorithm used in packet switched computer networks and telecommunications networks | https://en.wikipedia.org/wiki/Token_bucket |

Author: ```suggestion
| token bucket | 令牌桶 | An algorithm used in packet switched computer networks and telecommunications networks | https://en.wikipedia.org/wiki/Token_bucket |
```

> | token bucket | 令牌桶 | an algorithm used in packet switched computer networks and   telecommunications networks | https://en.wikipedia.org/wiki/Token_bucket |

Author: ```suggestion
| token bucket | 令牌桶 | an algorithm used in packet switched computer networks and telecommunications networks | https://en.wikipedia.org/wiki/Token_bucket |
```

| tolerate | 容忍 |  |  |
| topology | 拓扑结构 |  |  |
| traffic flow control | 流控/流量控制 |  |  |
| transaction | 事务 |  |  |
| traverse | 遍历 |  |  |
| trie | 字典树 |  |  |
| trie | 单词查找树 |  |  |
| TSO (Timestamp Oracle) | TSO (Timestamp Oracle) |  | https://pingcap.com/blog/Time-in-Distributed-Systems/#why-we-choose-tso |
| TTL (Time To Live) | 存活时间 |  |  |
| tutorial | 教程 |  |  |
| type | 类型 |  |  |
| type infer | 类型推导 |  |  |
| unbind | 解除（对服务，软件的）依赖，绑定，关联 |  |  |
| underlying table | 底层表 |  | https://dev.mysql.com/doc/refman/8.0/en/view-updatability.html |
| unidirectional data flow | 单向数据流 |  |  |
| union | 联合体 |  |  |
| unsigned integer | 无符号整数 |  |  |
| update | 修改（如：行） |  |  |
| user identification | 用户标识 |  |  |
| user privilege | 用户权限 |  |  |
| UUID (Universally Unique Identifier) | 通用唯一识别码 |  |  |
| variable | 变量 |  |  |
| verification | 验证 |  |  |
| verify | 验证 |  |  |

> | video transcoding |  |  |  |

Reviewer 2: ```suggestion
```

Reviewer 2: This word is not used in TiDB docs.

---
title: TiDB 中英术语表
summary: 本文档介绍 TiDB 中英文术语表。
---

> 本文提供了 TiDB 中英文术语对照表。

Reviewer 2: ```suggestion
# TiDB 中英术语表
```

| aggregate expression | 聚合表达式 |  | <https://www.ibm.com/support/knowledgecenter/en/SSGU8G_12.1.0/com.ibm.sqls.doc/ids_sqs_0189.htm> |
| aggregate function | 聚合函数 |  |  |
| alert | 报警 |  |  |
| algorithm | 算法 |  |  |
| anonymous | 匿名 |  |  |
| appendix | 附录 |  |  |
| architecture | 架构 |  |  |
| arithmetic operation | 数值运算/算术运算 |  |  |
| ascending order | 正序 |  |  |
| assignment expression | 赋值表达式 |  |  |
| assignment operation | 赋值运算 |  |  |
| authorize | 授予权限 |  |  |
| auto increment | 自增 |  |  |
| auto-concurrent/auto-concurrency | 自动线程数并行 |  |  |
| auto-failover | 故障自恢复 |  |  |
| auto-increment primary key | 自增主键 |  |  |
| availability | 可用性 |  |  |
| availability zone | 可用区 |  | <https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using-regions-availability-zones.html#concepts-regions-availability-zones> |
| backup | 备份 |  |  |
| bastion | 堡垒机 |  |  |
| binary literal | 二进制字面值 |  |  |
| bit | 位 |  |  |
| Bloom filter | 布隆过滤器 |  | <https://en.wikipedia.org/wiki/Bloom_filter> |
| boolean operation | 布尔运算 |  |  |

> | bootstrap | 启动\引导程序 |  |  |

Author: ```suggestion
| bootstrap | 启动/引导程序 |  |  |
```

| lock | 锁 |  |  |
| lock contention | 锁争用 |  |  |
| log file | 日志文件 |  |  |
| log level | 日志等级 |  |  |
| logic | 逻辑 |  |  |
| logic isolation | 逻辑隔离 |  |  |
| logical operation | 逻辑运算 |  |  |
| logical plan | 逻辑计划 |  |  |
| long transaction | 长事务 |  | <https://www.ibm.com/developerworks/cn/data/library/techarticles/dm-1001haodh/index.html> |
| machine code | 机器码 |  |  |
| massive data | 海量数据 | 不是 big data（大数据） |  |
| Massively Parallel Processing (MPP) | 大规模并行处理 |  |  |
| master | 主库 |  |  |
| materialized view | 物化视图 |  |  |
| memory | 内存 |  |  |
| merge join | 合并连接 |  |  |
| metadata | 元信息 |  |  |
| method | 方法 |  |  |
| method invocation | 方法调用 |  |  |
| metrics | 统计/指标（信息） |  |  |
| microservice | 微服务 |  |  |
| middleware | 中间件 |  |  |
| migration | 迁移 |  |  |
| module call | 模块调用 |  |  |
| monotone increasing | 单调递增 |  |  |
| mount option(s) | 挂载参数 |  |  |
| mount/mounting | 挂载 |  |  |
| multi nodes | 多节点 |  |  |
| multi-tenancy | 多租户技术 |  |  |
| mutation | 变异 |  | <http://www.orafaq.com/wiki/Mutating_table>、<https://blog.csdn.net/zhpsam109/article/details/97335> |
| mutex | 互斥锁 | Mutual exclusion | <https://zh.wikipedia.org/wiki/%E4%BA%92%E6%96%A5%E9%94%81> |

> | MVCC: multi-version concurrency control | 多版本并发控制 |  |  |

Author: ```suggestion
| MVCC (Multi-Version Concurrency Control) | 多版本并发控制 |  |  |
```

| MySQL syntax | MySQL 语法 |  |  |
| namespace | 命名空间 |  |  |
| native code | 原生码 |  |  |
| nest loop | 嵌套循环 |  |  |
| node | 节点 |  |  |
| non-unique index | 非唯一索引 |  |  |
| null | 空值 |  |  |
| object abstraction | 对象抽象 |  |  |
| offset | 偏移 |  |  |

> | OLAP: Online Analytical Processing | 在线分析处理 |  |  |
> | OLTP: Online Transactional Processing | 在线事务处理 |  |  |

Author: ```suggestion
| OLAP (Online Analytical Processing) | 在线分析处理 |  |  |
| OLTP (Online Transactional Processing) | 在线事务处理 |  |  |
```

| performance | 性能 |  |  |
| performance jitter | 性能抖动 |  |  |
| performance overhead | 性能开销 |  |  |
| performance tuning | 性能调优 |  |  |
| persist (data) | 持久化存储（数据） | persist 与一般的 save（保存）不一样，持久化存储是将数据保存到硬盘中，使得在应用程序或机器重启后可以继续访问之前保存的数据。 | <https://www.jianshu.com/p/6ce0864d2845> |
| persistent data | 持久化数据 |  |  |
| pessimistic locking | 悲观锁 |  |  |
| physical plan | 物理计划 |  |  |
| PoC (proof of concept) | 概念验证 |  |  |
| point query | 点查询 |  |  |
| populate | 填充 |  |  |
| port | 端口 |  |  |
| precheck | 前置检查 | TiDB 文档里统一译成“前置检查”，不译成”预检查” |  |
| prefix match | 前缀匹配 |  |  |
| primitive | 原语 | Kubernetes 中的概念 | <https://www.infoq.cn/article/69dzo9zwe5y3ca*5quqk> |
| private cloud | 私有云 |  |  |
| private key | 私钥 |  |  |
| privilege | 权限 |  |  |
| production environment | 生产环境 |  |  |
| public cloud | 公有云 |  |  |
| public key | 公钥 |  |  |
| pure virtual function | 纯虚函数 |  |  |
| push | 推送 |  |  |
| push down | 下推 | 动词短语 (verb phrase) |  |
| pushdown | 下推的 | 形容词 (adj.) |  |
| query | 查询 |  |  |

> | query by returning to the table (not a fixed term) | 回表查询 |  |  <https://odetocode.com/articles/237.aspx> |   |  |

Author: ```suggestion
| query by returning to the table (not a fixed term) | 回表查询 |  |  <https://odetocode.com/articles/237.aspx> |  
```

| read | 读 |  |  |
| reads and writes | 读写 |  |  |
| real-time replication delay | 实时同步延迟 |  |  |
| recursion | 递归 |  |  |
| reduction operation | 归约操作 |  |  |
| refactor | 重构 |  |  |
| Region | Region |  |  |
| regular expression | 正则表达式 |  |  |
| remote disaster recovery | 异地灾备 |  |  |
| renderer | 渲染器 |  |  |
| repair data | 修复数据 |  |  |
| replica | 副本 |  |  |
| replicator | 复制因子 |  |  |
| repository | 仓库 |  |  |
| request | 请求 |  |  |
| restart | 重启 |  |  |
| restore | 恢复 | 可以作为名词 (noun) 或动词 (verb) 使用 | <https://docs.microsoft.com/en-us/sql/relational-databases/backup-restore/back-up-and-restore-of-sql-server-databases?view=sql-server-ver15> |
| retention period | 保留时间 |  | <https://en.wikipedia.org/wiki/Retention_period> |
| return | 返回 |  |  |
| return value | 返回值 |  |  |
| reuse | 复用 |  |  |
| reverse proxy | 反向代理 |  |  |
| revoke | （权限）回收 |  |  |
| rewind | 回退 |  |  |
| rolling update | 滚动升级 |  | <https://kubernetes.io/docs/tutorials/kubernetes-basics/update-intro/> |
| rotate | 轮转 |  |  |
| rough set | 粗糙集 |  |  |
| routine | 例程 |  | <https://blog.csdn.net/scyatcs/article/details/77452336> |
| row store | 行存 |  |  |

> | RPC: remote procedure call | 远程过程调用 |  | <https://en.wikipedia.org/wiki/Remote_procedure_call> |

Author: ```suggestion
| Remote Procedure Call (RPC) | 远程过程调用 |  | <https://en.wikipedia.org/wiki/Remote_procedure_call> |
```

| target machine | （部署）目标机器 |  | <http://docs.ansible.com/ansible/latest/intro_inventory.html> |
| technology stack | 技术栈 |  |  |
| test | 测试 |  |  |
| the configuration file | 配置文件 |  |  |
| thread | 线程 |  |  |
| thread pool | 线程池 |  |  |
| throughput | 吞吐 |  |  |
| TiDB | TiDB |  |  |
| TiKV | TiKV |  |  |
| time out | 超时 |  |  |
| time series database | 时序数据库 |  |  |
| Time Series Database (TSDB) | 时序数据库 |  |  |
| time synchronization error | 时间同步误差 |  |  |
| timeliness | 时效性 |  |  |
| timestamp | 时间戳 |  |  |
| timestamp allocator | 时间戳分配器 |  |  |
| Timestamp Oracle | TSO |  |  |
| token bucket | 令牌桶 | An algorithm used in packet switched computer networks and telecommunications networks | <https://en.wikipedia.org/wiki/Token_bucket> |
| tolerate | 容忍 |  |  |
| topology | 拓扑结构 |  |  |
| traffic flow control | 流控/流量控制 |  |  |
| transaction | 事务 |  |  |
| traverse | 遍历 |  |  |
| trie | 字典树 |  |  |
| trie | 单词查找树 |  |  |
| TSO (Timestamp Oracle) | TSO (Timestamp Oracle) |  | <https://pingcap.com/blog/Time-in-Distributed-Systems/#why-we-choose-tso> |
| TTL (Time To Live) | 存活时间 |  |  |
| tutorial | 教程 |  |  |
| type | 类型 |  |  |
| type infer | 类型推导 |  |  |

> | unbind | 解除（对服务，软件的）依赖，绑定，关联 |  |  |

Author: ```suggestion
| unbind | 解除（对服务、软件的）依赖，绑定，关联 |  |  |
```

| underlying table | 底层表 |  | <https://dev.mysql.com/doc/refman/8.0/en/view-updatability.html> |
| unidirectional data flow | 单向数据流 |  |  |
| union | 联合体 |  |  |
| unsigned integer | 无符号整数 |  |  |

> | update | 修改（如：行） |  |  |

Author: ```suggestion
| update | 修改（如：修改行） |  |  |
```
